concordium-sdk/src/main/java/com/concordium/sdk/transactions/DeployModuleTransaction.java

```diff
@@ -0,0 +1,98 @@
+package com.concordium.sdk.transactions;
+
+import com.concordium.sdk.exceptions.TransactionCreationException;
+import com.concordium.sdk.transactions.smartcontracts.WasmModule;
+import com.concordium.sdk.types.UInt64;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+
+import java.util.Objects;
+
+/**
+ * A {@link DeployModuleTransaction} deploys compiled WASM smart contract module to chain.
+ */
+@Getter
+class DeployModuleTransaction extends AbstractTransaction {
+    /**
+     * A compiled Smart Contract Module in WASM with source and version.
+     */
+    private final WasmModule module;
+
+    private final AccountAddress sender;
+    private final AccountNonce nonce;
+    private final Expiry expiry;
+    private final TransactionSigner signer;
+    private final UInt64 maxEnergyCost;
```

Comment:
This field is in particular important to document as it specifies how much energy the transaction is allowed to use.